### PR TITLE
Add support for audit log entry events

### PR DESCRIPTION
**Pull request by FasterSpeeding**
### Checklist

- [ ] I have run `nox` and all the pipelines have passed.
- [ ] I have made unittests according to the code I have added/modified/deleted.

### CODE REVIEW

File: hikari/intents.py [hunk 1]
```diff
@@ -219,13 +219,17 @@ class Intents(enums.Flag):
...
+    GUILD_MODERATION = 1 << 2
     """Subscribes to the events listed below.
 
+    * `GUILD_AUDIT_LOG_ENTRY_CREATE`
     * `GUILD_BAN_ADD`
     * `GUILD_BAN_REMOVE`
     """
 
+    GUILD_BANS = GUILD_MODERATION
+    """Deprecated alias of `GUILD_BANS`."""
```
Comment (davfsa) [inline hikari/intents.py:231]: Can we do proper deprecation for this through a property?

File: hikari/impl/entity_factory.py [hunk 2]
```diff
@@ -767,61 +767,81 @@ def _deserialize_member_move_entry_info(
...
+        entries = {}
+        for entry_payload in payload["audit_log_entries"]:
+            try:
+                entry = self.deserialize_audit_log_entry(entry_payload, guild_id=guild_id)
+
+            except errors.UnrecognisedEntityError as exc:
+                _LOGGER.debug(exc.reason)
+
+            else:
+                entries[entry.id] = entry
```
Comment (davfsa) [inline hikari/impl/entity_factory.py:844]:
```suggestion
                entries[entry.id] = entry
 
            except errors.UnrecognisedEntityError as exc:
                _LOGGER.debug(exc.reason)
```

File: changes/1471.changes.md [hunk 3]
```diff
@@ -0,0 +1 @@
+Renamed `Intents.GUILD_BANS` to `Intents.GUILD_MODERATION`.
```
Comment (davfsa) [inline changes/1471.changes.md:1]: `.changes.md` doesnt exist. This would fall under `.deprecation.md`

**Review by davfsa — CHANGES_REQUESTED** [review]

**Comment by davfsa** [comment]
Thanks for working on this @FasterSpeeding!

File: hikari/impl/entity_factory.py (hunk 2 above)
Comment (FasterSpeeding) [reply hikari/impl/entity_factory.py:844]: I prefer limiting the scope of the try block to the bit I actually expect to raise when possible, tends to help with readability

Comment (davfsa) [reply hikari/impl/entity_factory.py:844]: What I dont like here is the lil jump that it does. In this case, the code isnt that big, and it should be quite clear that adding it to a dict shoudnt be raising a `UnrecognisedEntityError`

Comment (FasterSpeeding) [reply hikari/impl/entity_factory.py:844]: that jump is just the natural flow of a try, except, else statement though

File: hikari/intents.py (hunk 1 above)
Comment (FasterSpeeding) [reply hikari/intents.py:231]: Would require making a whole descriptor class for it since can't mix classmethod and property on 3.8

Comment (davfsa) [reply hikari/intents.py:231]: #1473 should aid with this :)

**Review by davfsa — APPROVED** [review]
LGTM. I added an extra commit to use `enum.deprecated` in the deprecated member and merged the master branch :)

**Review by davfsa — APPROVED** [review]

**Review by davfsa — APPROVED** [review]

**Pull request merged**